VSTSRestApiSamples/GettingStarted/Authentication.cs

```diff
@@ -1,55 +1,203 @@
+            {
+                throw ex.InnerException;
+            }
+
+            var bearerAuthHeader = new AuthenticationHeaderValue("Bearer", result.AccessToken);
+
+            return ListProjects(vstsAccountName, bearerAuthHeader);
+        }
+
+        //NOTE: If the user is not already logged in, this will cause a web browser prompt to display
+        public ListofProjectsResponse.Projects InteractiveADALExchangeGraphTokenForVSTSToken(string vstsAccountName, string applicationId)
+        {
+            AuthenticationContext ctx = GetAuthenticationContext(null);
+
+            AuthenticationResult result = null;
+            try
+            {
+                result = ctx.AcquireTokenAsync(GraphResourceId, applicationId, new Uri("urn:ietf:wg:oauth:2.0:oob"), new PlatformParameters(PromptBehavior.Auto)).Result;
+
+                //The result from the above call is now in the cache and will be used to assist in exchanging for a token given 
+                //a different resource ID
+                result = ctx.AcquireTokenSilentAsync(VSTSResourceId, applicationId).Result;
+            }
+            catch (Exception ex)
+            {
+                throw ex.InnerException;
+            }
+
+            var bearerAuthHeader = new AuthenticationHeaderValue("Bearer", result.AccessToken);
+
+            return ListProjects(vstsAccountName, bearerAuthHeader);
+        }
+
+        public ListofProjectsResponse.Projects NonInteractivePersonalAccessToken(string vstsAccountName, string personalAccessToken)
+        {
             // encode our personal access token                   
-            string credentials = Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", "", personalAccessToken)));
+            string encodedPAT = Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", "", personalAccessToken)));
+            var basicAuthHeader = new AuthenticationHeaderValue("Basic", encodedPAT);
+
+            return ListProjects(vstsAccountName, basicAuthHeader);
+        }
+
+        public ListofProjectsResponse.Projects DeviceCodeADAL(string vstsAccountName, string applicationId)
+        {
+            string tenant = GetAccountTenant(vstsAccountName);
+            AuthenticationContext ctx = GetAuthenticationContext(tenant);
+
+            AuthenticationResult result = null;
+            try
+            {
+                DeviceCodeResult codeResult = ctx.AcquireDeviceCodeAsync(VSTSResourceId, applicationId).Result;
+                Console.WriteLine("You need to sign in.");
+                Console.WriteLine("Message: " + codeResult.Message + "\n");
+                result = ctx.AcquireTokenByDeviceCodeAsync(codeResult).Result;
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+
+            var bearerAuthHeader = new AuthenticationHeaderValue("Bearer", result.AccessToken);
```

Comment:
These two lines are shared with both AAD flows. They should be refactored into a common method